test/opt/type_manager_test.cpp

```diff
@@ -175,6 +177,10 @@ std::vector<std::unique_ptr<Type>> GenerateAllTypes() {
   types.emplace_back(new RayQueryKHR());
   types.emplace_back(new HitObjectNV());
 
+  // SPV_AMDX_shader_enqueue
+  types.emplace_back(
+      new NodePayloadArrayAMDX(new Struct(std::vector<const Type*>{s32})));
```

Comment:
The struct that is allocated will be leaked. The pointer is passed to the `NodePayloadArrayAMDX`, but that class will not destroy it. You might want to reuse a type earlier in `types`, the same way the pointer types are handled.

Reply:
You want to use the one as line 137 I think.